thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/cache/RedisCacheDAO.java

```diff
@@ -0,0 +1,273 @@
+
+		int retryAttempts = ((Integer) dataSourceconfig.get(RETRY_ATTEMPTS) == null) ? DEFAULT_RETRY_ATTEMPTS
+				: (Integer) dataSourceconfig.get(RETRY_ATTEMPTS);
+		redissonConfig.useSingleServer().setRetryAttempts(retryAttempts);
+
+		int retryInterval = ((Integer) dataSourceconfig.get(RETRY_INTERVAL) == null) ? DEFAULT_RETRY_INTERVAL
+				: (Integer) dataSourceconfig.get(RETRY_INTERVAL);
+		redissonConfig.useSingleServer().setRetryInterval(retryInterval);
+
+		int pingConnectionInterval = ((Integer) dataSourceconfig.get(PING_CONNECTION_INTERVAL) == null)
+				? DEFAULT_PING_CONNECTION_INTERVAL
+				: (Integer) dataSourceconfig.get(PING_CONNECTION_INTERVAL);
+		redissonConfig.useSingleServer().setPingConnectionInterval(pingConnectionInterval);
+
+		boolean keepAlive = ((Boolean) dataSourceconfig.get(KEEP_ALIVE) == null) ? DEFAULT_KEEP_ALIVE
+				: (Boolean) dataSourceconfig.get(KEEP_ALIVE);
+		redissonConfig.useSingleServer().setKeepAlive(keepAlive);
+
+		boolean tcpNoDelay = ((Boolean) dataSourceconfig.get(TCP_NO_DELAY) == null) ? DEFAULT_TCP_NO_DELAY
+				: (Boolean) dataSourceconfig.get(TCP_NO_DELAY);
+		redissonConfig.useSingleServer().setKeepAlive(tcpNoDelay);
+
+		// Creating Redisson Client
+		redissonClient = getRedissonClient();
+		redisState = true;
+		LOG.info("Redis Connection Successful.");
+	}
+
+	/**
+	 * Fetches time-series data from cache. Formats it into a list of
+	 * TimeSeriesDataPoints before returning. The raw results will look something
+	 * like this: { { "time": 1000, "123456": "30.0" }, { "time": 2000, "123456":
+	 * "893.0" }, { "time": 3000, "123456": "900.6" } }
+	 * 
+	 * @param request ThirdEyeCacheRequest. Wrapper for ThirdEyeRequest.
+	 * @return list of TimeSeriesDataPoint
+	 */
+
+	public ThirdEyeCacheResponse tryFetchExistingTimeSeries(ThirdEyeCacheRequest request) throws Exception {
+		long metricId = request.getMetricId();
+		String dimensionKey = request.getDimensionKey();
+		String keyVal = createKey(metricId, dimensionKey);
+		RTimeSeries<Map<String, Object>> ts = getRedissonClient().getTimeSeries(keyVal);
+		ThirdeyeMetricsUtil.redisCallCounter.inc();
+		if (ts.size() == 0) {
+			LOG.info("no redis cache found for window startTime = {} to endTime = {}", request.getStartTimeInclusive(),
+					request.getEndTimeExclusive());
+			ThirdeyeMetricsUtil.redisExceptionCounter.inc();
+			return new ThirdEyeCacheResponse(request, new ArrayList<>());
+		}
+
+		long start = request.getStartTimeInclusive();
+		long end = request.getEndTimeExclusive() - request.getRequest().getGroupByTimeGranularity().toMillis();
+
+		Collection<Map<String, Object>> cacheValues = ts.range(start, end);
+		LOG.info("reading redis cache for window startTime = {} to endTime = {}", request.getStartTimeInclusive(),
+				request.getEndTimeExclusive());
+		List<TimeSeriesDataPoint> timeSeriesRows = new ArrayList<>();
+		for (Map<String, Object> cacheValue : cacheValues) {
+			long timestamp = (long) cacheValue.get(CacheConstants.TIMESTAMP);
```

Comment:
Where are you using the prefix created above when fetching the metrics?

Reply:
added comments. Comments filter 1 & 2 will be the fetches from the redis-store